docs/develop/gettingstarted.rst

```diff
@@ -0,0 +1,310 @@
+
+.. code-block:: console
+
+   $ sudo apt-get install build-essential libffi-dev git pkg-config
+
+For the stm32 port, the ARM cross-compiler is required:
+
+.. code-block:: console
+
+   $ sudo apt-get install arm-none-eabi-gcc arm-none-eabi-binutils arm-none-eabi-newlib
+
+See the `ARM GCC
+toolchain <https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm>`_
+for the latest details.
+
+Python 3 is also recommended:
+
+.. code-block:: console
+
+   $ python3 -v
+   Python 3.5.0 (default, Jul 17 2020, 14:04:10) 
+   [GCC 5.4.0 20160609] on linux
+   Type "help", "copyright", "credits" or "license" for more information.
+   >>> 
+
+All supported ports have different dependency requirements, see their respective
+`readme file <https://github.com/micropython/micropython/tree/master/ports>`_.
+
+Build the cross-compiler
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Almost all ports require building ``mpy-cross`` first to perform pre-compilation
+of scripts that will be included in the port firmware:
+
+.. code-block:: console
+
+   $ cd mpy-cross
+   $ make
+
+Also, ``mpy-cross`` must be built for the host architecture and not the target architecture.
+If it built successfully, you should see a message similar to this:
+
+.. code-block:: console
+
+   LINK mpy-cross
+      text	   data	    bss	    dec	    hex	filename
+    279328	    776	    880	 280984	  44998	mpy-cross
+
+.. note::
+
+   Use ``make -C mpy-cross`` to build the cross-compiler in one statement
+   otherwise, you will need to do ``cd ..`` for the next step.
+
+Build for the Unix Port
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To build for the unix port, make sure all linux related dependencies are 
+installed as detailed in the required dependencies section. Also, make sure you have a 
+working environment for ``gcc`` and ``GNU make``. Ubuntu 20.04 has been used for the example
+below but other unixes ought to work with little modification:
+
+.. code-block:: console
+
+   $ gcc --version
+   gcc (Ubuntu 9.3.0-10ubuntu2) 9.3.0
+   Copyright (C) 2019 Free Software Foundation, Inc.
+   This is free software; see the source for copying conditions.  There is NO
+   warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.then build:
+
+.. code-block:: console
+
+   $ cd ports/unix
+   $ make submodules
+   $ make
+
+If MicroPython built correctly, you should see the following:
+
+.. code-block:: console
+
+   LINK micropython
+      text	   data	    bss	    dec	    hex	filename
+    412033	   5680	   2496	 420209	  66971	micropython
+
+Now run it:
+
+.. code-block:: console
+
+   $ ./micropython
+   MicroPython v1.13-38-gc67012d-dirty on 2020-09-13; linux version
+   Use Ctrl-D to exit, Ctrl-E for paste mode
+   >>> print("hello world")
+   hello world
+   >>>
+
+Build for the stm32 port
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Like the unix port, you need to install some required dependencies
+as detailed in the required dependencies section, then build:
```

Comment:
Perhaps make "required dependencies" a link to the section above, and then specifically call out arm-none-eabi-gcc. Also note that it needs to be in the PATH or specified manually via CROSS_COMPILE.

(The reason is that the "required dependencies" section just tells you to install it, but doesn't verify that the build will know how to find it).